### ISSUE TITLE
Python dictConfig example config is inaccurate

### ISSUE THREAD

**Issue by FSund** [mCodingLLC/VideosSampleCode#17]
This is not the correct way to configure the root logger. This configures the logger named "root", which is a child of the actual root logger. The effect is that `logger = logging.getLogger("my_logger")` does not use the configuration from the JSON (since `"my_logger"` is not a child of `"root"`, but of the actual root logger, which has not been configured).

https://github.com/mCodingLLC/VideosSampleCode/blob/b6fe3c808c9a31b0425fe9f2005420fdb1958f0f/videos/135_modern_logging/logging_configs/2-stderr-json-file.json#L39-L47

The root logger uses a separate `root` key ([source](https://docs.python.org/3.8/library/logging.config.html#dictionary-schema-details)), so the correct JSON would be 

```JSON
  "loggers": {},
  "root": {
    "level": "DEBUG",
    "handlers": [
      "stderr",
      "file"
    ]
  }
```

(the empty `"loggers"` key can probably be skipped).

**Comment by harkabeeparolus** [comment]
Holy crap, I came here to say exactly this, 6 months later. 😳

**Comment by harkabeeparolus** [comment]
And here is the kicker... The incorrect configuration in the GitHub repo and in the video DOES WORK in Python >=3.9, but it fails (as documented) in Python 3.8... Which is what we had on one of our production systems. 🤯

**Comment by harkabeeparolus** [comment]
Here is more proof:

> [...] because the logger named `''` is the root logger. Specifying the top-level key `root` makes it clearer what you're doing if you're configuring a lot of loggers - the `''` logger configuration could be lost inside a group of others, whereas the `root` key is adjacent to the `loggers` key and so (in theory) should stand out more.
> 
> To reiterate, the key named `root` is a top-level key; it's not under `loggers`.

> 'root' is not a root logger, '' is the root logger.

https://stackoverflow.com/questions/20258986/root-logger-in-dictconfig

**Comment by harkabeeparolus** [comment]
Final update: mCoding responded to my comment on the [YouTube video](https://youtu.be/9L77QExPmI0):

> [...] using "root" instead of "" for the root logger wasn't introduced until Python 3.9: [What’s New In Python 3.9](https://docs.python.org/3/whatsnew/3.9.html) search "root". You're also right that the docs tell you to use a top level "root" key, although in a correct configuration there is no difference to putting it under loggers versus a top-level key. I hesitate to make the repo and video out of sync by moving the root key since this "Modern" video really assumes Python 3.11+ and 3.8 is EOL as of this month.

I agree with this opinion, and I suggest that this issue be closed after the compatibility warning was added in commit 4cd26933ededb6716f9cdf954365dd1a6f169344.

**Comment by Jos100703** [comment]
I use python 3.11 and the issue with having to name the root logger "" is still valid here.

**Comment by harkabeeparolus** [comment]
> I use python 3.11 and the issue with having to name the root logger "" is still valid here.

I don't believe that's the case. Please see my previous comment for more details. In my testing, I found the official Python documentation and changelog to be correct. Could you provide a minimum code sample that will reproduce the error on Python >= 3.9?

**Comment by Jos100703** [comment]
I spent 5 hours trying to fix this yesterday, and it only worked after I used "" instead "root". Now it works both ways (using the identical venv). Sometimes you just gotta hate python.
